snapper/task.py

```diff
@@ -25,16 +26,11 @@ def run(self, num_workers):
         if not Path(self.output_path).exists():
             os.makedirs(self.output_path)
 
-        capture_snaps(
-            urls=self.urls,
-            outpath=self.output_path,
-            timeout=self.timeout,
-            num_workers=num_workers,
-            user_agent=self.user_agent,
-            result=self.result,
-            phantomjs_binary=self.phantomjs_binary
-        )
-        self.status = "ready"
+        thread = threading.Thread(target=capture_snaps,
+          args = (self.urls, self.output_path, self.timeout, num_workers,
+                  self.user_agent, self.result, self.phantomjs_binary, self,)) 
+        thread.daemon = True
```

Comment:
That's generally not a really good idea, because you lose any control on when this thread will finish. I would suggest using something like this instead: https://docs.python.org/3/library/concurrent.futures.html#threadpoolexecutor